### ISSUE TITLE
Feature Request: Support harvester-baremetal-container-workload clusters

### ISSUE THREAD

**Issue by iamwillbar** [rancher/dashboard#12518]
### Is your feature request related to a problem?

There does not appear to be any way to target Harvester clusters when the harvester-baremetal-container-workload feature flag is enabled. Harvester clusters are excluded from the "All clusters in this workspace" filter, "Advanced" filter, or cluster groups that contain Harvester clusters.

### Solution you'd like

When the harvester-baremetal-container-workload feature flag is enabled, don't filter Harvester clusters from the target.

### Alternatives you've considered

Using advanced targeting and cluster groups.

**Comment by richard-cox** [comment]
If the UI supplies an explicit list it needs to take into account the `harvester-baremetal-container-workload` FF before filtering them out. We do this in a number of places, possibly not here.

FF Enable - don't filter harvester clusters
FF Disable (default) - filter harvester clusters

**Comment by torchiaf** [comment]
The UI adds a specific target to exclude Harvester clusters:

```
    - clusterSelector:
        matchExpressions:
          - key: provider.cattle.io
            operator: NotIn
            values:
              - harvester
```
But it doesn't take in account the feature flag, so we need to implement a check here.
We also need to verify that Harvester clusters are listed in Fleet Clusters table when target by any Fleet applications

**Comment by gaktive** [comment]
Related new ticket: https://github.com/rancher/rancher/issues/52309

**Comment by gaktive** [comment]
Target for this is Harvester 1.7.0 due this December so this would need to be in for 2.13.0 based on having this exposed for their GA of the feature. Need to figure out development time and QA requirements.

**Comment by richard-cox** [comment]
This request came in pretty close to 2.13.0 FC.

We could re-classify as a bug and get it in 2.13.1?

**Comment by gaktive** [comment]
From @rebeccazzzz, 2.13.1 & 2.14.0 are reasonable. The related backend ticket 52309 can be worked on by Harvester, though Rancher would need to review & merge. Harvester QA could then test.

**Comment by gaktive** [comment]
Goal now: 2.14.0 to align with Harvester 1.8.0. No backport at this time.

**Comment by richard-cox** [comment]
Transferring to @torchiaf to respond to https://github.com/rancher/dashboard/pull/14934#issuecomment-3431465660.

**Comment by richard-cox** [comment]
Notes for release note. This needs to go in the section that talks about `harvester-baremetal-container-workload` feature and bringing it out of experimental. If this doesn't exist yet ping us and we can discuss where to place

Something like...
"After enabling the `harvester-baremetal-container-workload` Feature Flag please validate existing Fleet Bundles to ensure they now select, or don't select, the required clusters."

**Comment by richard-cox** [comment]
Apparently GA for baremetal has now been [pushed to 2.15.0](https://github.com/rancher/rancher/issues/52309#issuecomment-3894032659). Think we should still aim for 2.14.0 unless there are any big blockers

**Comment by gaktive** [comment]
@richard-cox work with @marcelofukumoto and @nwmac to figure out what we can work with if truly, we aren't blocked by backend.

**Comment by richard-cox** [comment]
Chatted with Marcelo on Friday.

Not dependent on backend changes for this one, we just need to ensure our bugs are addressed by the time the feature becomes [GA](https://github.com/rancher/rancher/issues/52309).

As we've done some work on this already it would be nice to fix in 2.14.0, but has to come behind any existing 2.14.0 work.

**Comment by IsaSih** [comment]
Validated in 

[Rancher](https://github.com/rancher/rancher)	v2.15.0-alpha8
[Dashboard](https://github.com/rancher/dashboard)	v2.14.0-alpha7
[Rancher Prime CLI](https://github.com/rancher/cli)	v2.15.0-rc.1

with feature flag harvester-baremetal-container-workload enabled.

I'm reopening the ticket as in the PR it's listed as an acceptance criteria that "When harvester-baremetal-container-workload flag is enabled, the UI should support Harvester clusters as targets in App Bundles (HelmOps/GitRepos), Workspaces and Clusters pages.", AppBundles can only be linked to the harvester downstream cluster, not the imported ones, as seen in the screenshots:

<img width="1708" height="948" alt="Image" src="https://github.com/user-attachments/assets/3f1ecd9d-eb03-4673-8a77-11c184570766" />

<img width="1708" height="948" alt="Image" src="https://github.com/user-attachments/assets/4d530fa7-1e3a-4896-911a-bffd2258b75e" />

https://github.com/user-attachments/assets/28819b4b-c551-498e-919a-060850d413c0

https://github.com/user-attachments/assets/af6170dc-4809-4a1d-984b-a3f9faf8a105

**Comment by marcelofukumoto** [comment]
Hello @IsaSih ! 
Thank you for checking and reopening.
From what I could see from your test, you tested on dashboard `2.14`.
But the implementation is only on `2.15` for the UI to show on the cluster targets so you need to test using the dashboard on 2.15.

Maybe I am missing something, let me know and we can connect to check on it.

**Comment by IsaSih** [comment]
True, I realized the dashboard image was incorrect. Now repeating the tests with the following dashboard image (and same backend): 

[Rancher](https://github.com/rancher/rancher)	v2.15.0-alpha8
[Dashboard](https://github.com/rancher/dashboard)	master e1e6a88
[Rancher Prime CLI](https://github.com/rancher/cli)	v2.15.0-rc.1

With `harvester-baremetal-container-workload` feature flag enabled:
- Harvester Imported and downstream clusters are displayed in fleet clusters and fleet dashboard

<img width="1705" height="948" alt="Image" src="https://github.com/user-attachments/assets/b039d209-39d0-429a-b2e3-2aaad59f878f" />

-App Bundles can successfully target both harvester imported and downstream clusters targetting ALL or manual target

<img width="1708" height="948" alt="Image" src="https://github.com/user-attachments/assets/29b5ea25-0417-49a9-9fb0-425dcdc31c60" />

https://github.com/user-attachments/assets/6e47ba98-fafb-453e-a698-c8ca6d3ca9c3

- App Bundles were successfully deployed to both harvester imported and downstream clusters

https://github.com/user-attachments/assets/79b3b773-a0fd-4f53-a8d7-646523945e1e


- Cluster Groups with both harvester imported and downstream clusters were created

<img width="1705" height="948" alt="Image" src="https://github.com/user-attachments/assets/25bc77b0-e0e4-46db-b9ec-c3b7d4d2bd2d" />

<img width="1705" height="948" alt="Image" src="https://github.com/user-attachments/assets/86e51fba-c3a5-4c53-8462-2f0ae524414c" />

**Comment by IsaSih** [comment]
@marcelofukumoto could you please confirm if this is the expected behavior?
With `harvester-baremetal-container-workload` feature flag disabled:

Existing AppBundles that target both harvester imported and dowsntream clusters still have the original deployment count displayed in the UI and the app bundle resources for all of the old deployments. When you edit the app bundle, you cannot select the harvester imported clusters when target is all or manual, but they can still be target if the selector is based on cluster groups. And the Edit App Bundle form only displays the harvester downstream cluster

https://github.com/user-attachments/assets/b45cbc29-7716-4ed2-8c91-f1c9262a69f7

https://github.com/user-attachments/assets/68d3e9a0-b275-40c6-b78c-747d09a90b81

I'll address a separate bug related to the fleet UI not updating the resources count syncronously

**Comment by marcelofukumoto** [comment]
That is correct.
The previous select ones are still selected if you don't update the APP BUNDLE.
That means if you select them with the flag enabled, and than you disabled the flag. The APP BUNDLE will still have the same targets selected, including the harvester one.
If you try to update it will only remove if you remove it manually, or select the ALL option.

For GROUP it will ignore the flag completely. That is the exact same behavior on v2.14.

So the behaviours you shared are expected.

**Comment by IsaSih** [comment]
Thanks for confirming. I'll be closing the issue as expected then
